server/src/main/java/org/elasticsearch/search/fetch/FetchContext.java

```diff
@@ -207,4 +206,48 @@ public ScriptFieldsContext scriptFields() {
     public SearchExtBuilder getSearchExt(String name) {
         return searchContext.getSearchExt(name);
     }
+
+    /**
+     * Get the {@link MappedFieldType} for a field.
+     */
+    public MappedFieldType fieldType(String name) {
+        return searchContext.fieldType(name);
+    }
+
+    /**
+     * Returns all the fields that match a given pattern. If prefixed with a
+     * type then the fields will be returned with a type prefix.
+     */
+    public Set<String> simpleMatchToIndexNames(String pattern) {
+        return searchContext.simpleMatchToIndexNames(pattern);
+    }
+
+    public Set<String> sourcePaths(String name) {
+        return searchContext.sourcePaths(name);
+    }
+
+    /**
+     * @return Whether a field is a metadata field.
+     */
+    public boolean isMetadataField(String field) {
+        return searchContext.isMetadataField(field);
+    }
+
+    /**
+     * Is the source stored?
+     */
+    public boolean isSourceStored() {
+        return searchContext.isSourceStored();
+    }
+
+    public Analyzer indexAnalyzer() {
+        return searchContext.indexAnalyzer();
+    }
+
+    /**
+     * Returns the best nested {@link ObjectMapper} instances that is in the scope of the specified nested docId.
+     */
+    public ObjectMapper findNestedObjectMapper(int nestedDocId, IndexSearcher searcher, LeafReaderContext context) throws IOException {
+        return searchContext.findNestedObjectMapper(nestedDocId, searcher, context);
+    }
```

Comment:
this is where I stopped in my PR because it was getting bigger. I think although it takes longer, we should go incrementally with these changes. I was about to look into FetchContext next.